### ISSUE TITLE
Pre-preparation of triples

### ISSUE THREAD

**Issue by mortendahl** [tf-encrypted/tf-encrypted#337]
Sampling and sending triples takes up a significant amount of time for Pond/SecureNN. Test whether it makes a difference (after #137 has landed) to have these distributed ahead of time, ie in a proper offline phase. One concrete way of during this could be via [tf.FIFOQueue](https://www.tensorflow.org/api_docs/python/tf/FIFOQueue)s.

**Comment by mortendahl** [comment]
Closing in favour of #417
